### PR TITLE
fix(api): use token decimals when looking up market price

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David Adams <david@umaproject.org>

**Motivation**

https://app.clubhouse.io/uma-project/story/1963/bug-adjust-matcha-price-call-to-use-correct-decimals

**Summary**

Some market prices were not showing, this may have been to querying the matcha api with the wrong quantity of synthetic. Small change to use token decimals to set the correct synthetic amount when calling matcha api. 

**Details**

This synthetic token used to not return a market price: 0x2dE7A5157693a895ae8E55b1e935e23451a77cB3, which it now does after this fix 

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

https://app.clubhouse.io/uma-project/story/1963/bug-adjust-matcha-price-call-to-use-correct-decimals

### CODE REVIEW

File: packages/api/src/services/market-prices.ts [hunk 1]
```diff
@@ -72,7 +73,7 @@ export default function (config: Config, appState: Dependencies) {
   // we can try to price all known erc20 addresses. Some will fail. Also this endpoint does not return a timestamp
   // so we will just set one from our query time.
   async function update(timestampS = nowS()) {
-    const addresses = Array.from(collateralAddresses.values()).concat(Array.from(syntheticAddresses.values()));
+    const addresses = Array.from(new Set(Array.from(collateralAddresses).concat(Array.from(syntheticAddresses))));
```
Comment (daywiss) [inline packages/api/src/services/market-prices.ts:76]: make sure we dont somehow have duplicate addresses

Comment (mrice32) [reply packages/api/src/services/market-prices.ts:76]: nit: for readability, should we use lodash uniq? https://lodash.com/docs/4.17.15#uniq

**Review by mrice32 — APPROVED** [review]
LGTM!

**Comment by daywiss** [comment]
> nit: for readability, should we use lodash uniq? https://lodash.com/docs/4.17.15#uniq

Yea i can use that. I considered it actually but decided to use standard libs, but it would be cleaner with lodash

**Pull request merged**